src/services/refactors/convertExport.ts

```diff
@@ -165,6 +165,7 @@ namespace ts.refactor {
         const checker = program.getTypeChecker();
         const exportSymbol = Debug.checkDefined(checker.getSymbolAtLocation(exportName), "Export name should resolve to a symbol");
         FindAllReferences.Core.eachExportReference(program.getSourceFiles(), checker, cancellationToken, exportSymbol, exportingModuleSymbol, exportName.text, wasDefault, ref => {

```

Comment:
There's a lot of code backing `eachExportReference`, but my superficial impression is that it's expected to return a bunch of imports.  If that's the case, why would `exportName` ever be in the list?  Assuming that was unintentional, should the filtering go in `eachExportReference` in case it ever gets another caller?

Reply:
@a-tarasyuk Ping

Reply:
@amcasey Do you mean that self-reference names should be skipped in `eachExportReference`?

Reply:
Well, I don't know the intended behavior of `eachExportReference`, but intuitively I would expect it to only return imports.  If it consistently returns augmentations as well and that's important, then filtering at the call site seems reasonable.

Reply:
If that's the case, should the filter drop everything that's not an import and not just self-references?

Reply:
`importTracker` returns `indirectUsers` which includes self-referenced export. Not sure if this is expected behavior or not.
https://github.com/microsoft/TypeScript/blob/29c7ae23cadd3ef866ebddc3923423eec27fc531/src/services/findAllReferences.ts#L1186

Reply:
`importTracker` returns `indirectUsers` which includes self-referenced export. Not sure if this is expected behavior or not.
https://github.com/microsoft/TypeScript/blob/29c7ae23cadd3ef866ebddc3923423eec27fc531/src/services/findAllReferences.ts#L1186

Reply:
Right, but it filters them

https://github.com/microsoft/TypeScript/blob/751c11412618e8b2a504380135bc2f0b2505e18f/src/services/findAllReferences.ts#L1199

I was wondering whether the check actually belonged there.